docs/user/security/secure-saved-objects.asciidoc

```diff
@@ -0,0 +1,90 @@
+
+[IMPORTANT]
+============================================================================
+Bulk key rotation can consume a considerable amount of resources and hence only user with a `superuser` role can trigger it.
+============================================================================
+
+[source,sh]
+--------------------------------------------------
+$ curl -X POST /api/encrypted_saved_objects/_rotate_key
+--------------------------------------------------
+// KIBANA
+
+The API returns the following:
+
+[source,sh]
+--------------------------------------------------
+{
+  "total": 1000, <1>
+  "successful": 300, <2>
+  "failed": 0 <3>
+}
+--------------------------------------------------
+
+<1> The total number of _all_ Saved Objects with the encrypted data, irrespective to the key {kib} used for encryption.
+<2> The number of the Saved Objects that were still encrypted with one of the old encryption keys and that {kib} successfully re-encrypted with the primary key.
+<3> The number of the Saved Objects that were still encrypted with one of the old encryption keys that {kib} failed to re-encrypt with the primary key.
+
+It's also possible to rotate encryption key only for a specific type of the Saved Objects using an optional `type` query string parameter (default: all types), or/and specify a max batch size with an optional `batchSize` query string parameter (default and max: 10000):
+
+[NOTE]
+============================================================================
+We optimized default parameters for speed and don't expect you to change any of those without a strong need. However, if you're experiencing any issues with this API you may want to decrease a batch size or rotate encryption key for the specific types only. In this case you may need to run key rotation multiple times in a row.
+============================================================================
+
+[source,sh]
+--------------------------------------------------
+$ curl -X POST /api/encrypted_saved_objects/_rotate_key?type=alert&batchSize=5000 <1>
+--------------------------------------------------
+// KIBANA
+
+<1> {kib} will rotate encryption key only for the Saved Objects with the `alert` type in batches of 5000 objects.
+
+For the time being if you specify decryption-only keys {kib} may try to use them all if it cannot decrypt the Saved Object using the primary encryption key. In most of the cases this overhead is negligible, but if you're constantly dealing with the thousands of the Saved Objects that are encrypted and experiencing performance issues you may want to try to rotate encryption key for those objects.
```

Comment:
__note:__ I'm torn on this one, maybe it's more confusing than useful?

Reply:
I feel like this would be better suited at the top of the API doc (once extracted, see other comment). This does a nice job of explaining why you might want to use this API, so I think it's worthwhile to keep in.

Reply:
Yep, moved, thanks!